### PR TITLE
Update JWT logic

### PR DESCRIPTION
**Pull request by pawbu**
- use JWKS endpoint to get the public key
- audience in QA different from PROD

### CODE REVIEW

File: data_service/api/auth.py [hunk 1]
```diff
@@ -1,14 +1,27 @@
...
+
+jwks_client = PyJWKClient(os.environ['JWKS_URL'], lifespan=3000)
+
+
+def get_jwks_aud() -> str:
+    return "datastore-qa" if os.environ['STACK'] == 'qa' else "datastore"
+
+
+def get_signing_key(jwt_token: str):
+    signing_key = jwks_client.get_signing_key_from_jwt(jwt_token)
```
Comment (DanielElisenberg) [inline data_service/api/auth.py:20]: Can just return directly

File: data_service/api/auth.py [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
 import jwt
+from fastapi import HTTPException, status
+from jwt import PyJWKClient
 from jwt.exceptions import (
     InvalidSignatureError, ExpiredSignatureError,
     InvalidAudienceError, DecodeError
 )
-from fastapi import HTTPException, status
+
+jwks_client = PyJWKClient(os.environ['JWKS_URL'], lifespan=3000)
```
Comment (DanielElisenberg) [inline data_service/api/auth.py:12]: Didn't know about this one! Very handy!

File: tests/unit/api/test_auth.py [hunk 3]
```diff
@@ -62,18 +64,17 @@ def test_auth_missing_token():
...
-    assert "Internal Server Error" in e.value.detail
-
-
 def test_auth_toggled_off(monkeypatch):
     monkeypatch.setenv('JWT_AUTH', 'false')
     user_id = authorize_user(None)
     assert user_id == "default"
+
+
+def test_jwt_audience_qa(monkeypatch):
```
Comment (DanielElisenberg) [inline tests/unit/api/test_auth.py:73]: :100:

File: tests/unit/api/test_data_api.py [hunk 4]
```diff
@@ -50,8 +51,10 @@ def get_processor_override():
 
 @pytest.fixture(autouse=True)
 def setup(monkeypatch):
```
Comment (DanielElisenberg) [inline tests/unit/api/test_data_api.py:53]: not important, but pytest added public exports for the MonkeyPatch type

```py
from pytest import MonkeyPatch

def test_my_thing(monkeypatch: MonkeyPatch):
    monkeypatch. # autocompletion and docs available now due to type
```

File: tests/unit/core/test_processor.py [hunk 5]
```diff
@@ -85,7 +86,7 @@ def test_invalid_fixed_request():
     )
 
 
-def teardown_function(file_name):
```
Comment (DanielElisenberg) [inline tests/unit/core/test_processor.py:88]: :100:

**Review by DanielElisenberg — APPROVED** [review]
Looks great! :100:

File: tests/unit/api/test_data_api.py (hunk 4 above)
Comment (pawbu) [reply tests/unit/api/test_data_api.py:53]: That is nice!

**Pull request merged**